monitor_state_dict.py

```diff
@@ -0,0 +1,110 @@
+"""
+Contains the definitions for monitor.state.
+
+This is essentially a stopgap while OOP state is worked on.
+"""
+from __future__ import annotations
+
+from typing import MutableMapping, TYPE_CHECKING, Any, DefaultDict, Dict, List, Literal, Optional, Set, Tuple, TypedDict
+
+
+class MonitorStateDict(TypedDict):
+    """Top level state dictionary for monitor.py."""
+
+    # Game related
+    GameLanguage:           Optional[str]               # From `Fileheader`
+    GameVersion:            Optional[str]               # From `Fileheader`
+    GameBuild:              Optional[str]               # From `Fileheader`
+    Horizons:               Optional[bool]                        # Does the player have Horizons?
+    Odyssey:                bool                        # Have we detected Odyssey?
+
+    # Multi-crew
+
+    Captain:                Optional[str]               # If on a crew, the captian's name
+    Role:                   Optional[Literal['Idle', 'FireCon', 'FighterCon']]  # Role in crew
+
+    # Cmdr state
+    FID:                    Optional[str]               # Frontier CMDR ID
+    Friends:                Set[str]                    # Online Friends
+    Credits:                int
+    Loan:                   Optional[int]
+    Engineers:              Dict[Any, Any]              # TODO
+    Rank:                   Dict[Any, Any]              # TODO
+    Reputation:             Dict[Any, Any]              # TODO
+    Statistics:             Dict[Any, Any]              # This is very freeform.
+
+    # Engineering Materials
+    Raw:                    DefaultDict[str, int]
+    Encoded:                DefaultDict[str, int]
+    Manufactured:           DefaultDict[str, int]
+
+    # Ship
+    ShipID:                 Optional[str]
+    ShipIdent:              Optional[str]
+    ShipName:               Optional[str]
+    ShipType:               Optional[str]
+
+    HullValue:              Optional[int]
```

Comment:
For all three of `HullValue`, `ModulesValue` and `Rebuy` we should be OK switching to a default of `0`.  It's falsey, so `if <var>` checks won't change.  We *will* need to specify a non-default default on `entry.get(...)` calls to use `0` instead of `None`.